server/utils/flow/handler.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package flow
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/httputils"
+)
+
+type fh struct {
+	flow  Flow
+	store FlowStore
+}
+
+func Init(h *httputils.Handler, flow Flow, store FlowStore) {
+	fh := &fh{
+		flow:  flow,
+		store: store,
+	}
+
+	oauth2Router := h.Router.PathPrefix("/").Subrouter()
+	oauth2Router.HandleFunc(flow.URL(), fh.handleFlow).Methods("POST")
+}
+
+func (fh *fh) handleFlow(w http.ResponseWriter, r *http.Request) {
+	mattermostUserID := r.Header.Get("Mattermost-User-ID")
+	if mattermostUserID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	stepNumber, err := strconv.Atoi(r.URL.Query().Get("step"))
+	if err != nil {
+		fmt.Println("Step is not an int, err=" + err.Error())
+		http.Error(w, "Step is not an int, err="+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	step := fh.flow.Step(stepNumber)
+	if step == nil {
+		fmt.Println("Not valid step")
+		http.Error(w, "Not valid step", http.StatusInternalServerError)
+		return
+	}
+
+	property := step.PropertyName()
+
+	valueString := r.URL.Query().Get(property)
+	if valueString == "" {
+		fmt.Println("Correct property not set")
+		http.Error(w, "Correct property not set", http.StatusInternalServerError)
+		return
+	}
+
+	value := valueString == "true"
+
+	err = fh.store.SetProperty(mattermostUserID, property, value)
+	if err != nil {
+		fmt.Println("There has been a problem setting the property, err=" + err.Error())
+		http.Error(w, "There has been a problem setting the property, err="+err.Error(), http.StatusInternalServerError)
```

Comment:
I suggest having `fh` take a `Logger` so we can log this properly. Also put the error string in a var in the line above to be used in `Logger.Errorf` and `http.Error`, to avoid duplication.

Reply:
This was actually some lines for myself for debugging that slipped into the commit. Would you still like these errors to be logged in the server?

Reply:
It doesn't seem too critical to me. 0/5 remove the debug print